src/whylogs/core/summaryconverters.py

```diff
@@ -118,6 +121,41 @@ def single_quantile_from_sketch(sketch: kll_floats_sketch, quantile: float):
     return type("Object", (), {"quantile": qval[0]})
 
 
+def _calculate_bins(end: float, start: float, n: int, avg_per_bucket: float, max_buckets: int):
+    # Include the max value in the right-most bin
+    end += abs(end) * 1e-7
+    abs_end = abs(end)
+    abs_start = abs(start)
+    max_magnitude = max(abs_end, abs_start)
+
+    # the kll_floats_sketch use 32bit floats, so we check precision against np.float32
+    float_mantissa_bits = np.finfo(np.float32).nmant
+
+    # Include the right edge in the bin edges
+    n_buckets = min(math.ceil(n / avg_per_bucket), max_buckets)
+    width = (end - start) / n_buckets
+
+    # check for precision of width with respect to float_mantissa_bits and bin width:
+    bits_in_max = math.floor(math.log2(max_magnitude))
+    width_bits = math.floor(math.log2((end - start) / n_buckets))
+    logger.debug(f"bits_in_max is: {bits_in_max}")
+    logger.debug(f"width_bits is: {width_bits}")
+
+    if bits_in_max - float_mantissa_bits > width_bits:
```

Comment:
```
# If the bin width is smaller than min_interval, we need bigger bins
if width < min_interval:
    new_buckets = math.floor((end - start) / min_width
    n_buckets = max(new_buckets, 1)
    width = (end - start) / n_buckets

Reply:
This considers the bin widths rather than the bits in resolution, maybe this is easier to read, done.